### PR TITLE
Fixed inconsistent whitespace and added .editorconfig file

### PR DESCRIPTION
**Pull request by jamestmartin**
[EditorConfig](https://editorconfig.org/) is a widely-supported editor configuration file that specifies basic information about coding styles (in particular regarding the whitespace). While writing my other commits, I noticed a lot of issues with indentation consistency (trailing whitespace after lines, sometimes an end-of-file newline but sometimes not, some files were indented with 2 spaces and others with 4, and there were even a few tabs scattered around); this motivated me to create a `.editorconfig` file to reduce the quantity of these inconsistencies introduced in the future.

I additionally moved the JavaScript files written by external sources from `assets/` to `vendor/`, which is what `vendor/` is for. This change is purely organizational and shouldn't affect the way these files are served. In addition to being more semantically correct, this also allowed me to exclude these files from the local editorconfig style options.

Finally, with the assistance of an automated tool, I corrected the inconsistent whitespace throughout the project. I manually corrected any undesirable changes that the tool made. Special attention was paid to `.slim` and `.erb` files, because those files can produce different results based on trailing whitespace. I cannot guarantee that no mistakes were added to those files, but I hope that your review can catch any (and if that's too sketchy for you, worst case I can just undo the changes to those files-- but I hope it won't come to that).

I did *not* go through the effort of manually fixing the whitespace in the test files (beyond that the automated tool did), but I will later. See the comment below.

### CODE REVIEW

**Comment by jamestmartin** [comment]
Once #87 is merged I will fix the indentation in the test cases. Many of the tests are indented with four spaces rather than two, but I ignored this previously because it didn't matter because the tests weren't working. Now I have to wait as to avoid merge conflicts.

**Comment by jamestmartin** [comment]
I can confirm that other already-extant PRs are going to cause merge conflicts for this PR. Don't worry about those; I'll take care of them myself. That said, I'd still appreciate it if you'll review this PR as-is anyway so you can merge it sooner once those merge conflicts are solved (and the sooner this PR is merged, the less work to fix merge conflicts I have to do).

**Comment by jamestmartin** [comment]
In the most recent commit I applied not only the formatting fixes, but also other automatic rubocop fixes. Based on manual review, in my opinion, this looks good, but you should probably see what you think about it.

However, there were two questionable changes in my opinion:

1. The `# frozen_string_literal: true` comments. These feel more like something that should probably be enabled project-wide rather than per-file. On the other hand, trying to enable it project-wide could potentially break things, and as far as incremental changes go, the tests are pretty much the best place to start (because it doesn't affect prod and is, by definition, immediately tested).

2. The use of single quotes in the test names. At least some of the spec test case names contain single quotes, so they still use double quotes in the names. This may be fine, but it may also be better to keep consistency, with either double quotes everywhere, or renaming the tests not to use single quotes in the name (the better choice IMO). Though that's just bikeshedding/nitpicking.

The automatic fixes took care of all of the rubocop warnings in the spec tests and most of them in the standard tests, except for a few line length issues that I'm not sure how to re-format idiomatically. Also, I didn't use the `rubocop-rails` or `rubocop-rspec` linters because they're not enabled in this branch. I'm alright with that, though, because I need to be focusing on actual code and features rather than obsessing over fixing up tiny style issues right now, and those sorts of changes are out of the scope of this commit.

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
+# frozen_string_literal: true
+
 ENV['RAILS_ENV'] ||= 'test'
-require File.expand_path('../../config/environment', __FILE__)
+require File.expand_path('../config/environment', __dir__)
```
Comment (the1banana) [inline spec/spec_helper.rb:4]: What's the significance of this change?

Comment (jamestmartin) [reply spec/spec_helper.rb:4]: rubocop did it.

**Pull request merged**